lib/core/src/main/java/io/atlasmap/core/DefaultAtlasContext.java

```diff
@@ -604,25 +601,10 @@ private void processSourceFieldMappings(DefaultAtlasSession session, List<Field>
                 return;
             }
 
-            if (sourceField.getIndex() != null) {
-                AtlasPath path = new AtlasPath(sourceField.getPath());
-                if (path.hasCollection() && !path.isIndexedCollection()) {
-                    if (sourceField.getIndex() != 0) {
-                        AtlasUtil.addAudit(session, sourceField.getDocId(),
-                            String.format("Index on a field is not supported, if there is a collection without index on the path",
-                                sourceField.getPath()), sourceField.getPath(), AuditStatus.ERROR, null);
```

Comment:
We actually have to support it now since expressions require indexed parameters and we allow to pass a collection without index to e.g. concatenate all items.